coral-component-shell/src/scripts/ShellSelectListSwitcher.js

```diff
@@ -0,0 +1,112 @@
+/**
+ * Copyright 2021 Adobe. All rights reserved.
+ * This file is licensed to you under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License. You may obtain a copy
+ * of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under
+ * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+ * OF ANY KIND, either express or implied. See the License for the specific language
+ * governing permissions and limitations under the License.
+ */
+
+import {BaseComponent} from '../../../coral-base-component';
+import {SelectableCollection} from '../../../coral-collection';
+import selectListSwitcher from '../templates/selectListSwitcher';
+import {SelectList} from '../../../coral-component-list';
+import {commons} from '../../../coral-utils';
+
+const CLASSNAME =  '_coral-Shell-selectListSwitcher';
+
+/**
+ @class Coral.Shell.SelectListSwitcher
+ @classdesc A Shell SelectList Switcher component
+ @htmltag coral-shell-selectlistswitcher
+ @extends {HTMLElement}
+ @extends {BaseComponent}
+ */
+class ShellSelectListSwitcher extends BaseComponent(HTMLElement) {
+  /** @ignore */
+  constructor() {
+    super();
+
+    // Template
+    this._elements = {};
+
+    selectListSwitcher.call(this._elements);
+
+    this._delegateEvents({
+      'coral-selectlist:change':'_onSelectListChange',
+      'coral-shell-switcherlist-item:change': '_onUpdateHref'
```

Comment:
When is this event coral-shell-switcherlist-item:change triggered

Reply:
whenever there is a change in attribute of coral-shell-switcherlist-item, this is used to update the selectlistitem attribute accordingly